deploy/stacks/lambda_api.py

```diff
@@ -91,14 +91,18 @@ def __init__(
         self.api_handler_dlq = self.set_dlq(f'{resource_prefix}-{envname}-graphql-dlq')
         api_handler_sg = self.create_lambda_sgs(envname, 'apihandler', resource_prefix, vpc)
         api_handler_env = {'envname': envname, 'LOG_LEVEL': 'INFO', 'REAUTH_TTL': str(reauth_ttl)}
+        # Check if custom domain exists and if it exists email notifications could be enabled. Create a env variable which stores the domain url. This is used for sending data.all share weblinks in the email notifications.
+        if custom_domain and custom_domain.get('hosted_zone_name', None):
+            api_handler_env['frontend_domain_url'] = f'https://{custom_domain.get("hosted_zone_name", None)}'
+        api_handler_role = self.create_function_role(envname, resource_prefix, 'graphql', pivot_role_name)
```

Comment:
I think we do not need to create the `api_handler_role` outside of the custom_auth if-statement and we can keep the previous implementation `role=self.create_function_role(envname, resource_prefix, 'graphql', pivot_role_name),` in line 105

Reply:
Fixed!!